components/browser/session/src/main/java/mozilla/components/browser/session/SelectionAwareSessionObserver.kt

```diff
@@ -44,6 +44,19 @@ abstract class SelectionAwareSessionObserver(
         activeSession?.register(this)
     }
 
+    /**
+     * Starts observing changes to the session matching the [sessionId].
+     *
+     * @param sessionId the session ID to observe.
+     */
+    fun observeFromId(sessionId: String?) {
```

Comment:
Since this method does a lot more than expected, i.e. "if the session with an ID doesn't exist, observe the selected session.", we could name it `observeIdOrSelected`?

`observeFromId` signifies, it will _only_ observe the session with the ID.

Reply:
Agreed. `observeIdOrSelected` is better :)